src/definitions.jl

```diff
@@ -396,6 +396,49 @@ function ifftshift(x,dim)
     circshift(x, s)
 end
 
+##############################################################################
+
+
+struct Frequencies{T<:Number} <: AbstractVector{T}
+    nreal::Int
+    n::Int
+    multiplier::T
```

Comment:
The constructor should maybe have some sanity checks that `1 ≤ nreal ≤ n`?

Reply:
Now we `error` if the condition isn't satisfied.

Is there a more specific error that we could throw instead? Maybe a `DomainError`?

Reply:
Usually this would be `ArgumentError`